Firestore/core/src/firebase/firestore/local/leveldb_migrations.cc

```diff
@@ -261,8 +262,16 @@ void LevelDbMigrations::RunMigrations(leveldb::DB* db) {
 
 void LevelDbMigrations::RunMigrations(leveldb::DB* db,
                                       SchemaVersion to_version) {
-  LevelDbTransaction transaction{db, "Read schema version"};
-  SchemaVersion from_version = ReadSchemaVersion(&transaction);
+  SchemaVersion from_version = ReadSchemaVersion(db);
+  // If this is a downgrade, just save the downgrade version so
+  // detect it when we go to upgrade again, allowing us to rerun the
+  // data migrations.
+  if (from_version > to_version) {
+    LevelDbTransaction transaction(db, "Save downgrade version");
```

Comment:
Under the former transaction arrangement we could have used a single transaction to read the version and write the downgrade version. Is it worth trying to conserve transaction creation?

Normally I don't think this would matter, but we do these on every startup, and even though we're not blocking the main thread, these still contribute to the latency to first query.

Note that we can preserve the nice change to the tests by adding a local override in the tests that instantiates the transaction and then reads using the production code that doesn't.

Reply:
We could have, but we never did, and probably wouldn't. Some migrations run iteratively and commit periodically, so we don't run them all in a single transaction.